public/app/features/variables/state/actions.ts

```diff
@@ -517,7 +518,9 @@ export const variableUpdated = (
     const variables = getVariables(state);
     const g = createGraph(variables);
     const panels = state.dashboard?.getModel()?.panels ?? [];
-    const affectedPanelIds = getAllAffectedPanelIdsForVariableChange(variableInState.id, variables, panels);
+    const affectedPanelIds = isAdHoc(variableInState)
+      ? undefined // for adhoc variables we don't know which panels that will be impacted
+      : getAllAffectedPanelIdsForVariableChange(variableInState.id, variables, panels);
```

Comment:
would this be better as `[]` instead of `undefined`? then the type is consistent with what `getAllAffectedPanelIdsForVariableChange` returns. 🤔

Reply:
You're right it would be better to have an empty array, but then instead of `if(affectedPanelIds)` it would be `if(affectedPanelIds.length)` everywhere. I don't have any strong opinions about this except I would opt for doing that in a separate PR. WDYT @ashharrison90?

Reply:
gotcha, that's fine. i didn't think about the case where we were checking for existence, was only thinking about the loop cases (`affectedPanelIds.forEach`, etc.) which would fail with `undefined`.